src/Compilers/CSharp/Portable/Symbols/PublicModel/MethodSymbol.cs

```diff
@@ -0,0 +1,322 @@
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class MethodSymbol : Symbol, IMethodSymbol
+    {
+        private readonly Symbols.MethodSymbol _underlying;
+        private ITypeSymbol _lazyReturnType;
+        private ImmutableArray<ITypeSymbol> _lazyTypeArguments;
+        private ITypeSymbol _lazyReceiverType;
+
+        public MethodSymbol(Symbols.MethodSymbol underlying)
+        {
+            Debug.Assert(underlying is object);
+            _underlying = underlying;
+        }
+
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+        internal Symbols.MethodSymbol UnderlyingMethodSymbol => _underlying;
+
+        MethodKind IMethodSymbol.MethodKind
+        {
+            get
+            {
+                switch (_underlying.MethodKind)
+                {
+                    case MethodKind.AnonymousFunction:
+                        return MethodKind.AnonymousFunction;
+                    case MethodKind.Constructor:
+                        return MethodKind.Constructor;
+                    case MethodKind.Conversion:
+                        return MethodKind.Conversion;
+                    case MethodKind.DelegateInvoke:
+                        return MethodKind.DelegateInvoke;
+                    case MethodKind.Destructor:
+                        return MethodKind.Destructor;
+                    case MethodKind.EventAdd:
+                        return MethodKind.EventAdd;
+                    case MethodKind.EventRemove:
+                        return MethodKind.EventRemove;
+                    case MethodKind.ExplicitInterfaceImplementation:
+                        return MethodKind.ExplicitInterfaceImplementation;
+                    case MethodKind.UserDefinedOperator:
+                        return MethodKind.UserDefinedOperator;
+                    case MethodKind.BuiltinOperator:
+                        return MethodKind.BuiltinOperator;
+                    case MethodKind.Ordinary:
+                        return MethodKind.Ordinary;
+                    case MethodKind.PropertyGet:
+                        return MethodKind.PropertyGet;
+                    case MethodKind.PropertySet:
+                        return MethodKind.PropertySet;
+                    case MethodKind.ReducedExtension:
+                        return MethodKind.ReducedExtension;
+                    case MethodKind.StaticConstructor:
+                        return MethodKind.StaticConstructor;
+                    case MethodKind.LocalFunction:
+                        return MethodKind.LocalFunction;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(_underlying.MethodKind);
+                }
```

Comment:
>} [](start = 16, length = 1)

Can this be reduced to `=> _underlying.MethodKind`?

If not, can we merge the `case` blocks?
```
case MethodKind.AnonymousFunction:
case ...:
case MethodKind.LocalFunction:
    return _underlying.MethodKind;
default:
    ...
```

Reply:
> Can this be reduced to => _underlying.MethodKind?

This is the original implementation of this API. Changing it is outside of scope of this PR. Feel free to open an issue. 

---
In reply to: [339177034](https://github.com/dotnet/roslyn/pull/39498#discussion_r339177034) [](ancestors = 339177034)